cleanup_utils/integration_cleanup.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# OpenCraft -- tools to aid developing and hosting free software projects
+# Copyright (C) 2015-2018 OpenCraft <contact@opencraft.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Integration cleanup script
+
+Cleans up all AWS, Openstack and DNS resources left behind by CircleCI
+cancelled runs
+"""
+
+import argparse
+from datetime import datetime, timedelta
+import logging
+import os
+from pytz import UTC
+
+from cleanup_utils.aws_cleanup import AwsCleanupInstance
+from cleanup_utils.openstack_cleanup import OpenStackCleanupInstance
+from cleanup_utils.dns_cleanup import DnsCleanupInstance
+
+
+# Constants ###################################################################
+
+DEFAULT_AGE_LIMIT = datetime.utcnow().replace(tzinfo=UTC) - timedelta(days=3)
+
+
+# Logging #####################################################################
+
+logger = logging.getLogger('integration_cleanup')
+
+file_handler = logging.FileHandler('integration_cleanup.log')
+file_handler.setLevel(logging.DEBUG)
+
+console_handler = logging.StreamHandler()
+console_handler.setLevel(logging.DEBUG)
+
+logger.addHandler(file_handler)
+logger.addHandler(console_handler)
+
+
+# Classes #####################################################################
+
+
+def main():
+    """
+    Main function responsible for the cleanup
+    """
+    parser = argparse.ArgumentParser(
+        description="Cleanup all unused resources from integration runs that "
+                    "have been cancelled or failed."
+    )
```

Comment:
@giovannicimolin, this script should cleanup all old instances irrespective of whether they were created in cancelled or failed or successful runs.